### PR TITLE
fix: 1154 Add LOGOUT_REDIRECT_URL setting

### LINKED ISSUE #1154: Make redirect after logout configurable
Hi all,

I think it would be really useful if the URL that Flask-Appbuilder redirects to after logout would be configurable - akin to Django's `LOGOUT_REDIRECT_URL` setting.

Right now, the user is redirected to the hompage:

[flask_appbuilder/security/views.py#L477](https://github.com/dpgaspar/Flask-AppBuilder/blob/master/flask_appbuilder/security/views.py#L477)
```
class AuthView(BaseView):
    route_base = ""
    login_template = ""
    invalid_login_message = lazy_gettext("Invalid login. Please try again.")
    title = lazy_gettext("Sign In")

    @expose("/login/", methods=["GET", "POST"])
    def login(self):
        pass

    @expose("/logout/")
    def logout(self):
        logout_user()
        return redirect(self.appbuilder.get_url_for_index)
```

This might not be what is desired though, maybe the user should see a "thank you and see you soon page" or something like that, and it can also lead to problems with setups that use oauth2 for authentication, because the user will be logged straight back in if the oauth2 provider is configured to allow login without a prompt. I think this is what happens in #1110
So I think having an option to configure where users are redirected to after logout would be really useful, e.g.

```
    return redirect(
        self.appbuilder.app.config.get(
            "LOGOUT_REDIRECT_URL", self.appbuilder.get_url_for_index
        )
    )
```

### PR DESCRIPTION
**Pull request by blag**
### Description

This pull request addresses #1154, by redirecting to `LOGOUT_REDIRECT_URL` in the `/logout` view if `LOGOUT_REDIRECT_URL` is set in the config, falling back to `self.appbuilder.get_url_for_index`.

It also adds `LOGOUT_REDIRECT_URL` to the **Configuration keys** table in `docs/config.rst`, and adds a unit test.

This PR is the exact same changes as #1160, but reopened since that one didn't seem to garner any feedback or interest once it was closed by the stale bot. This PR will cleanly merge, and will likely also solve apache/airflow#17279, so there is already people waiting on this fix to be merged.

### ADDITIONAL INFORMATION

- [x] Has associated issue: fixes #1154 (which was automatically closed by the stalebot because it was "stale", even though the issue was not fixed)
- [ ] Is CRUD MVC related.
- [x] Is Auth, RBAC security related.
- [ ] Changes the security db schema.
- [ ] Introduces new feature
- [ ] Removes existing feature

### CODE REVIEW

File: docs/config.rst [hunk 1]
```diff
@@ -237,6 +237,8 @@ Use config.py to configure the following parameters. By default it will use SQLL
 |                                        | the existing languages with the countries  |           |
 |                                        | name and flag                              |           |
 +----------------------------------------+--------------------------------------------+-----------+
+| LOGOUT_REDIRECT_URL               | The location to redirect to after logout   |   No      |
++-----------------------------------+--------------------------------------------+-----------+
```
Comment (jedcunningham) [inline docs/config.rst:241]:
```suggestion
+----------------------------------------+--------------------------------------------+-----------+
| LOGOUT_REDIRECT_URL                    | The location to redirect to after logout   |   No      |
+----------------------------------------+--------------------------------------------+-----------+
```

Comment (blag) [reply docs/config.rst]: Thanks!

**Comment by blag** [comment]
Commenting so the stalebot doesn't mark this as stale. This PR is in a mergable state, it just needs a review.
@dpgaspar - Are you going to have time to review and merge this? Or do you need some additional help managing this project?

**Comment by blag** [comment]
Thank you!

**Pull request merged**